src/System.Windows.Extensions/src/System/Drawing/FontConverter.cs

```diff
@@ -126,94 +123,179 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
+                        if ((fontStyle | validBits) != validBits)
+                        {
+                            throw new InvalidEnumArgumentException(nameof(styleStr), (int)fontStyle, typeof(FontStyle));
+                        }
+                    }
                 }
             }
 
-            f_style = FontStyle.Regular;
-            if (fields.Length > 2)
-            {   // We have style
-                string compare;
+            return new Font(fontName, fontSize, fontStyle, units);
+        }
 
-                for (int i = 2; i < fields.Length; i++)
-                {
-                    compare = fields[i];
+        private (string, string) ParseSizeTokens(string text, char separator)
+        {
+            string size = null;
+            string units = null;
 
-                    if (compare.IndexOf("Regular") != -1)
-                    {
-                        f_style |= FontStyle.Regular;
-                    }
-                    if (compare.IndexOf("Bold") != -1)
-                    {
-                        f_style |= FontStyle.Bold;
-                    }
-                    if (compare.IndexOf("Italic") != -1)
-                    {
-                        f_style |= FontStyle.Italic;
-                    }
-                    if (compare.IndexOf("Strikeout") != -1)
-                    {
-                        f_style |= FontStyle.Strikeout;
-                    }
-                    if (compare.IndexOf("Underline") != -1)
+            text = text.Trim();
+
+            int length = text.Length;
+            int splitPoint;
+
+            if (length > 0)
+            {
+                // text is expected to have a format like " 8,25pt, ". Leading and trailing spaces (trimmed above), 
+                // last comma, unit and decimal value may not appear.  We need to make it ####.##CC
+                for (splitPoint = 0; splitPoint < length; splitPoint++)
+                {
+                    if (char.IsLetter(text[splitPoint]))
                     {
-                        f_style |= FontStyle.Underline;
+                        break;
                     }
                 }
+
+                char[] trimChars = new char[] { separator, ' ' };
+
+                if (splitPoint > 0)
+                {
+                    size = text.Substring(0, splitPoint);
+                    // Trimming spaces between size and units.
+                    size = size.Trim(trimChars);
+                }
+
+                if (splitPoint < length)
+                {
+                    units = text.Substring(splitPoint);
+                    units = units.TrimEnd(trimChars);
+                }
             }
 
-            return new Font(font, f_size, f_style, f_unit);
+            return (size, units);
+        }
+
+        private GraphicsUnit ParseGraphicsUnits(string units)
+        {
+            GraphicsUnit fontSizeUnit;
+            switch (units)
```

Comment:
this was previously case insensitive. now it's not. intentional?
https://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/FontConverter.cs,495

Reply:
I did this to match the current .NET Core behavior for parsing units. https://source.dot.net/#System.Windows.Extensions/System/Drawing/FontConverter.cs,152
Not sure what our final policy should be here

Reply:
OK that's fine